lnwallet/wallet.go

```diff
@@ -461,41 +481,61 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 		// Coin selection is done on the basis of sat/kw, so we'll use
 		// the fee rate passed in to perform coin selection.
 		var err error
-		selected, err = l.selectCoinsAndChange(
-			req.FundingFeePerKw, req.LocalFundingAmt, req.MinConfs,
-			req.SubtractFees,
+		fundingReq := &chanfunding.Request{
+			RemoteAmt:    req.RemoteFundingAmt,
+			LocalAmt:     req.LocalFundingAmt,
+			MinConfs:     req.MinConfs,
+			SubtractFees: req.SubtractFees,
+			FeeRate:      req.FundingFeePerKw,
+			ChangeAddr: func() (btcutil.Address, error) {
+				return l.NewAddress(WitnessPubKey, true)
+			},
+		}
+		fundingIntent, err = req.ChanFunder.ProvisionChannel(
+			fundingReq,
 		)
 		if err != nil {
+			if fundingIntent != nil {
```

Comment:
assumption  should be that `fundingIntent` is always nil whenever an error is returned.

Reply:
Hmm, yeah that's a safe assumption, removed.